docs/articles/documentation/recipes/best-practices/create-faster-tests.md

```diff
@@ -0,0 +1,101 @@
+
+TestCafe's [speed](../../reference/command-line-interface.md#--speed-factor) option allows you to change test speed. The default value is `1`, which is fastest.
+If your tests have `speed` set in the run configuration, disable this setting or set it to `1`.
+
+## Run Tests in Headless Browsers
+
+Run tests in headless browsers. Headless browsers take less time to initialize because they don't need to render the application to a GUI. They also enable you to run tests in environments that lack GUI capabilities, like CI containers.
+
+To run tests in a headless browser, use the `:headless` CLI parameter:
+
+```sh
+testcafe "chrome:headless" tests/
+```
+
+For more information, see [Test in Headless Mode](../../guides/concepts/browsers.md#test-in-headless-mode)
+
+## Use Roles for Login
+
+If your tests require activity from logged in users, use [Roles](../../guides/advanced-guides/authentication.md#user-roles) for authentication.
+
+Put Roles in the [beforeEach hook](../../reference/test-api/fixture/beforeeach.md) of your test suite: following tests in a suite reuse authentication data and authentication happens instantly.
+
+With `Roles`, TestCafe remembers cookies associated with every logged-in account. When you switch Roles, TestCafe erases authentication data and you don't have to log out manually.
+
+If your test covers logins on multiple websites, the active Role collects the authentication data from all these websites. When you switch to that Role later, you are logged in to all these websites.
+
+Use an [Anonymous Role](../../guides/advanced-guides/authentication.md#anonymous-role) to instantly log out of all accounts during a test.
+
+> Roles can access authentication data in cookie and browser storage. If your authentication system stores data elsewhere, you may not be able to use roles.
+
+For more info on Roles, see [User Roles](../../guides/advanced-guides/authentication.md#user-roles)
+
+## Run Tests Concurrently
+
+Tests execute faster if run concurrently. In concurrent mode, TestCafe creates multiple instances of specified browsers and uses that browser pool to run tests.
+
+Use the [--concurency](../../reference/command-line-interface.md#-c-n---concurrency-n) CLI option to launch tests concurrently:
+
+```sh
+testcafe --concurency 3 chrome tests/
+```
+
+For more info on concurrency, read [Run Tests Concurrently](../../guides/basic-guides/run-tests.md#run-tests-concurrently).
+
+## Mock Requests
+
+A tested application may interact with remote resources (for example, an analytics service or a database). Requests to such resources may create delays in test suite execution. To avoid such delays, you can mock requests to these resources.
+
+TestCafe [request mocker](../../reference/test-api/requestmock/README.md) intercepts requests from your app to external resources and responds with data that you specify. A mocked request is resolved almost instantly, which eliminates a possible delay caused by a foreign resource or network lag.
```

Comment:
minor stylistic edits.
foreign is a bad word in this context, too imprecise. (also see: alien). an external resource is not necessarily foreign (for example if we're talking about a different server on the same CDN)
```suggestion
TestCafe [request mocker](../../reference/test-api/requestmock/README.md) intercepts your application's requests to external resources, and responds with the data that you specify. A mocked request is resolved almost instantly, which eliminates a possible delay caused by data processing and network lag.
```

Reply:
I like `latency` better

```suggestion
TestCafe [request mocker](../../reference/test-api/requestmock/README.md) intercepts your application's requests to external resources, and responds with the data that you specify. A mocked request is resolved almost instantly, which eliminates a possible delay caused by data processing and network latency.
```